.eleventy.js

```diff
@@ -159,25 +159,27 @@ module.exports = function (eleventyConfig) {
 
   eleventyConfig.addCollection('postsByTag', (collectionApi) => {
     const tagMap = new Map()
-
     const posts = collectionApi.getAll()
 
     for (const post of posts) {
       if (post.data.tags) {
         for (const tag of post.data.tags) {
           const key = tag.toLowerCase()
 
-          // Initialize group if not already
           if (!tagMap.has(key)) {
             tagMap.set(key, [])
           }
 
-          // Add post to tag group
           tagMap.get(key).push(post)
         }
       }
     }
 
+    // Sort posts for each tag by descending date
+    for (const postList of tagMap.values()) {
+      postList.sort((a, b) => new Date(b.data.date) - new Date(a.data.date))
+    }
```

Comment:
If you need to add comments then it be best to refactor it out to a method that encapsulate the comment.

Something along the lines of...
```javascript
function sortPostsByDescendingDate(tagMap) {
  for (const postList of tagMap.values()) {
    postList.sort((a, b) => new Date(b.data.date) - new Date(a.data.date));
  }
}
```